ape/ape.lds

```diff
@@ -478,6 +485,17 @@ HIDDEN(ape_stack_filesz = 0);
 HIDDEN(ape_stack_memsz = STACKSIZE);
 HIDDEN(ape_stack_align = 16);
 
+HIDDEN(ape_loader_offset = ape_rom_offset + ape_loader - ape_rom_vaddr);
```

Comment:
You wouldn't need to define an ELF segment for the loader payload. All the linker script needs to do is something like this:

```
HIDDEN(ape_loader_b0 = RVA(ape_loader) % 10 + 48);
HIDDEN(ape_loader_b1 = RVA(ape_loader) < 10 ? 32 : RVA(ape_loader) / 10 % 10 + 48);
HIDDEN(ape_loader_b2 = RVA(ape_loader) < 100 ? 32 : RVA(ape_loader) / 100 % 10 + 48);
HIDDEN(ape_loader_b3 = RVA(ape_loader) < 1000 ? 32 : RVA(ape_loader) / 1000 % 10 + 48);
HIDDEN(ape_loader_b4 = RVA(ape_loader) < 10000 ? 32 : RVA(ape_loader) / 10000 % 10 + 48);
HIDDEN(ape_loader_b5 = RVA(ape_loader) < 100000 ? 32 : RVA(ape_loader) / 100000 % 10 + 48);
HIDDEN(ape_loader_b6 = RVA(ape_loader) < 1000000 ? 32 : RVA(ape_loader) / 1000000 % 10 + 48);
HIDDEN(ape_loader_b7 = RVA(ape_loader) < 10000000 ? 32 : RVA(ape_loader) / 10000000 % 10 + 48);
HIDDEN(ape_loader_b8 = RVA(ape_loader) < 100000000 ? 32 : RVA(ape_loader) / 100000000 % 10 + 48);
HIDDEN(ape_loader_b9 = RVA(ape_loader) < 1000000000 ? 32 : RVA(ape_loader) / 1000000000 % 10 + 48);
```

You'd also want to define a section... It should probably go right before `.zip` in the linker script, e.g.

```ld-script
  .payload . : {
    KEEP(*(.payload))
  }
```

Reply:
Isn't what `SHSTUB2` is for? Apart from that, I can remove other definitions.

Will do the separate `.payload ` section. I was just not really sure this would not mess up the computation of the offset inside the file, but that should be fine as long as I deal with LOADADDR, right?